src/serialization/json_object.cpp

```diff
@@ -1111,7 +1112,7 @@ void fromJsonValue(const rapidjson::Value& val, cryptonote::rpc::BlockHeaderResp
   GET_FROM_JSON_OBJECT(val, response.reward, reward);
 }
 
-void toJsonValue(rapidjson::Writer<epee::byte_stream>& dest, const rct::rctSig& sig)
+void toJsonValue(rapidjson::Writer<epee::byte_stream>& dest, const rct_sig sig)
```

Comment:
If you introduce locals for the actual (`rct::rctSig`) `sig` and `pruned` values, and change the name of the param here to reflect that it's now a wrapper struct, the diff would be a lot smaller, and better reflect the nature of the change, as well as the underlying procedure (there's only one `sig.sig`, and it's the heart of the procedure, so it's semantically odd to continually use indirection).

Reply:
That's true, I can move all of the signature output code into the parent function (is that you are saying). The diff size would be the same, because I had to rename everything in the sub-function anyway.

Reply:
Oh never mind, I see what you mean. Yeah, I can call the code directly instead of using `INSERT_INTO_JSON` ?

Reply:
FWIW I partially did it this way because my new serialization routines will basically require it be done like this.